utils/ixdtf/src/lib.rs

```diff
@@ -177,12 +214,71 @@
 //! assert_eq!(offset.hour, 1);
 //! ```
 //!
-//! ## Additional DateTime grammar resources
+//! #### Implementing Annotation Handlers
+//!
+//! As mentioned in the prior section, there may be times where an application may
+//! need to implement application defined behavior for user defined functionality.
+//! In this instance, `ixdtf` provides a `*_with_annotation_handler` method that
+//! allows to the user to provide a callback.
+//!
+//! A handler is defined as `handler: impl FnMut(Annotation<'a>) -> Option<Annotation<'a>>`
+//! where `ixdtf` provides visibility to an annotation to the user. The call to this callback
+//! occurs prior to the `ixdtf`'s processing of the annotation, and will only occur if the
+//! annotation is provided back to `ixdtf`.
+//!
+//! If the user wishes to ignore any `ixdtf`'s errors, then they may return `None`, which
+//! results in a no-op for that annotation.
+//!
+//! Unless the user’s application has a specific reason to bypass action on an annotation,
+//! such as, custom unknown key handling or superceding a calendar based on it’s critical
+//! flag, it is recommended to return the annotation value.
+//!
+//! ##### Handler Example
+//!
+//! A user may wish to implement a custom key in an annotation set. This can be completed
+//! with custom handler.
+//!
+//! ```rust
+//! use ixdtf::parsers::IxdtfParser;
+//!
+//! let example_with_custom_key = "2024-03-02T08:48:00-05:00[u-ca=iso8601][!answer-to-universe=fortytwo]";
+//!
+//! let mut answer = None;
+//!
+//! let _ = IxdtfParser::new(example_with_custom_key).parse_with_annotation_handler(|annotation| {
+//!     if annotation.key == "answer-to-universe" {
+//!         answer = Some(annotation);
```

Comment:
Suggestion: since this is supposed to show you how to write one of these handlers correctly, please use `get_or_insert` to retain the first value. Alternatively, implement the additional logic involving mismatched critical annotations.